### ISSUE TITLE
Link not rendered properly

### ISSUE THREAD

**Issue by mholt** [yuin/goldmark#284]
goldmark has [https://github.com/yuin/goldmark/discussions](Discussions) in github.
You should post only issues here. Feature requests and questions should be posted at discussions.

- [x] goldmark is fully compliant with the CommonMark. Before submitting issue, you **must** read [CommonMark spec](https://spec.commonmark.org/0.29/) and confirm your output is different from [CommonMark online demo](https://spec.commonmark.org/dingus/).
    - [x] **Extensions(Autolink without `<` `>`, Table, etc) are not part of CommonMark spec.** You should confirm your output is different from other official renderers correspond with an extension.
- [x] **goldmark is not dedicated for Hugo**. If you are Hugo user and your issue was raised by your experience in Hugo, **you should consider create issue at Hugo repository at first** .

Please answer the following before submitting your issue:

1. What version of goldmark are you using? : v1.4.4
2. What version of Go are you using? : 1.17.7
3. What operating system and processor architecture are you using? : linux amd64
4. What did you do? :

This Markdown:

```markdown
[`route`](route) wraps other directives like `handle` does, but with two distinctions: 1) route blocks are not mutually exclusive to each other, and 2) directives within a route are not [re-ordered](/docs/caddyfile/directives#directive-order), giving you more control if needed.
```

5. What did you expect to see? :

HTML:

```html
<p><a href="route"><code>route</code></a> wraps other directives like <code>handle</code> does, but with two distinctions: 1) route blocks are not mutually exclusive to each other, and 2) directives within a route are not <a href="/docs/caddyfile/directives#directive-order">re-ordered</a>, giving you more control if needed.</p>
```

AST:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE document SYSTEM "CommonMark.dtd">

<document xmlns="http://commonmark.org/xml/1.0">
  <paragraph>
    <link destination="route" title="">
      <code>route</code>
    </link>
    <text> wraps other directives like </text>
    <code>handle</code>
    <text> does, but with two distinctions: 1) route blocks are not mutually exclusive to each other, and 2) directives within a route are not </text>
    <link destination="/docs/caddyfile/directives#directive-order" title="">
      <text>re-ordered</text>
    </link>
    <text>, giving you more control if needed.</text>
  </paragraph>
</document>
```

8. What did you see instead? :

```
<a href="[route](view-source:https://caddyserver.com/docs/caddyfile/directives/route)"><code>route</code></a> wraps other directives like <code>handle</code> does, but with two distinctions: 1) route blocks are not mutually exclusive to each other, and 2) directives within a route are not [re-ordered](<a href="[/docs/caddyfile/directives#directive-order](view-source:https://caddyserver.com/docs/caddyfile/directives#directive-order)">directive order</a>, giving you more control if needed.
```

10. Did you confirm your output is different from [CommonMark online demo](https://spec.commonmark.org/dingus/) or other official renderer correspond with an extension?:

Yes.

Basically, the link with text "re-ordered" is not rendered properly, but it is on the CommonMark demo.

We [discovered this issue on the Caddy docs](https://caddyserver.com/docs/caddyfile/directives/handle) and it is locally reproducible: https://github.com/caddyserver/website/issues/217

As an aside, we really love using Goldmark in Caddy to render web page Markdown. Thank you!

**Comment by mholt** [comment]
Indeed, v1.4.8 fixes it. (I looked at the commits but didn't see anything obvious there that would have solved it. Mea culpa.)
